service/header/sync.go

```diff
@@ -383,46 +367,48 @@ func (s *Syncer) processHeaders(ctx context.Context, from, amount uint64) (uint6
 		return 0, err
 	}
 
+	// update the state before ProvideHeights below to always keep state in sync with heightSub
+	s.stateLk.Lock()
+	s.state.Height = to
+	s.stateLk.Unlock()
+
 	s.heightSub.ProvideHeights(ctx, headers...)
 	return uint64(len(headers)), nil
 }
 
-// getHeaders gets headers from either remote peers or from local cache of headers received by PubSub
-func (s *Syncer) getHeaders(ctx context.Context, start, amount uint64) ([]*ExtendedHeader, error) {
-	// short-circuit if nothing in pending cache to avoid unnecessary allocation below
-	if _, ok := s.pending.FirstRangeWithin(start, start+amount); !ok {
-		return s.exchange.RequestHeaders(ctx, start, amount)
+// TODO(@Wondertan): Number of headers that can be requested at once. Either make this configurable or,
+//  find a proper rationale for constant.
+// TODO(@Wondertan): Make configurable
```

Comment:
can you make it a config param when constructing the syncer?

Reply:
In a separate PR

Reply:
ack

Reply:
https://github.com/celestiaorg/celestia-node/issues/449